contracts/governance/utils/Votes.sol

```diff
@@ -29,18 +29,26 @@ import "./IVotes.sol";
  * _Available since v4.5._
  */
 abstract contract Votes is IVotes, Context, EIP712 {
-    using Checkpoints for Checkpoints.History;
+    using Checkpoints for Checkpoints.Trace224;
     using Counters for Counters.Counter;
 
     bytes32 private constant _DELEGATION_TYPEHASH =
         keccak256("Delegation(address delegatee,uint256 nonce,uint256 expiry)");
 
     mapping(address => address) private _delegation;
-    mapping(address => Checkpoints.History) private _delegateCheckpoints;
-    Checkpoints.History private _totalCheckpoints;
+    mapping(address => Checkpoints.Trace224) private _delegateCheckpoints;
+    Checkpoints.Trace224 private _totalCheckpoints;
 
     mapping(address => Counters.Counter) private _nonces;
 
+    /**
+     * @dev Clock used for flagging checkpoints. Can be overridden to implement timestamp based checkpoints (and voting).
+     * See EIP-5805.
+     */
+    function clock() public view virtual returns (uint256) {
```

Comment:
We need to add `CLOCK_MODE`. Not mandatory but recommended by the EIP ("SHOULD").

Are we planning to offer a timestamp extension? We could consider a contract `ClockTimestamp` implementing the two clock functions which could be combined with `Votes` and `ERC20Votes`, although the overrides aren't very clean (because it's not chaining via super).

Reply:
I'm not sure how the inheritance tree between `ClockTimestamp` and `Votes` would work. Maybe we will need a `Clock` interface ... but interfaces have only external and we need clock to be public